### ISSUE TITLE
libpcap filters doesn't work on pypy

### ISSUE THREAD

**Issue by gkonstantyno** [secdev/scapy#2814]
Socket initialization with filter raises
TypeError: unsupported operand type(s) for +: 'str' and 'StructureMeta'

This stopped working with version 2.4.4.
Compiling filters was moved from tcpdump to libpcap

#### Brief description

```
>>> s = conf.L2socket(filter="udp")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/opt/pypy/site-packages/scapy/arch/linux.py", line 407, in __init__
    attach_filter(self.ins, filter, iface)
  File "/opt/pypy/site-packages/scapy/arch/linux.py", line 143, in attach_filter
    sock.setsockopt(socket.SOL_SOCKET, SO_ATTACH_FILTER, bp)
  File "/opt/pypy/lib_pypy/_ctypes/structure.py", line 309, in __buffer__
    fmt = get_format_str(self)
  File "/opt/pypy/lib_pypy/_ctypes/array.py", line 309, in get_format_str
    ch = get_format_str(obj)[1:]
  File "/opt/pypy/lib_pypy/_ctypes/array.py", line 322, in get_format_str
    return byteorder[sys.byteorder] + ch
TypeError: unsupported operand type(s) for +: 'str' and 'StructureMeta'
```

#### Environment

-   Scapy version: 2.4.4
-   Python version: PyPy 7.3.1
-   Operating System: Debian buster

#### How to reproduce

This can be reproduced with docker pypy image

docker run --rm -it pypy:3 /bin/bash
> apt update & apt install libpcap-dev
> pip install scapy
> scapy
>>> s = conf.L2socket(filter="udp")

**Comment by gpotter2** [comment]
This is a PyPy bug. You should probably create a issue on https://foss.heptapod.net/pypy/pypy/-/issues
There was already a bug on this VERY issue which they fixed 10months ago, and already managed to break again: https://foss.heptapod.net/pypy/pypy/-/issues/3114 . I'm slightly tired of it

**Comment by gpotter2** [comment]
Apparently this will be fixed in 7.3.2, thanks for the report
